test/graceful-shutdown.test.js

```diff
@@ -1,54 +1,54 @@
 'use strict'
 
-const t = require('tap')
-// Tests skip on win32 platforms due SIGINT signal is not supported across all windows platforms
-const test = (process.platform === 'win32') ? t.skip : t.test
+const { test, beforeEach, afterEach } = require('node:test')
+const assert = require('assert')
 const sinon = require('sinon')
 const start = require('../start')
 
 let _port = 3001
-
 function getPort () {
   return '' + _port++
 }
 
-let spy = null
 let fastify = null
+let spy = null
 let signalCounter = null
 const sandbox = sinon.createSandbox()
 
-t.beforeEach(async () => {
+// Skip tests on Windows
+const isWindows = process.platform === 'win32'
+const conditionalTest = isWindows ? test.skip : test
+
+beforeEach(async () => {
   signalCounter = process.listenerCount('SIGINT')
 
   const argv = ['-p', getPort(), './examples/plugin.js']
   fastify = await start.start(argv)
   spy = sinon.spy(fastify, 'close')
 })
 
-t.afterEach(async () => {
+afterEach(async () => {
   sandbox.restore()
 })
 
-test('should add and remove SIGINT listener as expected ', async t => {
-  t.plan(2)
-
-  t.equal(process.listenerCount('SIGINT'), signalCounter + 1)
+conditionalTest('should add and remove SIGINT listener as expected', async (t) => {
+  assert.strictEqual(process.listenerCount('SIGINT'), signalCounter + 1)
 
   await fastify.close()
 
-  t.equal(process.listenerCount('SIGINT'), signalCounter)
-
-  t.end()
+  assert.strictEqual(process.listenerCount('SIGINT'), signalCounter)
 })
 
-test('should have called fastify.close() when receives a SIGINT signal', async t => {
-  process.once('SIGINT', () => {
-    sinon.assert.called(spy)
-
-    t.end()
-
-    process.exit()
-  })
-
+conditionalTest('should call fastify.close() on SIGINT', async (t) => {
+  const sigintHandler = () => {
+    try {
+      sinon.assert.called(spy)
+      t.pass('fastify.close() was called on SIGINT')
```

Comment:
I think we should check this with a `t.plan`.
Especially because `t.pass` is not part of the node test runner, so this test should fail

Reply:
The test fails if I put the method "plan" 
<img width="885" height="449" alt="screenshot-error-plan" src="https://github.com/user-attachments/assets/e7659f42-3ac8-4e2d-95ec-1460f1e8c69c" />